test/types/scalar/vass/int8toReal32.good

```diff
@@ -1,2 +1,10 @@
+int8toReal32.chpl:22: warning: potentially surprising implicit conversion from 'int(8)' to 'real(32)'
+int8toReal32.chpl:22: note: add a cast :real(32) to avoid this warning
+int8toReal32.chpl:23: warning: potentially surprising implicit conversion from 'uint(8)' to 'real(32)'
+int8toReal32.chpl:23: note: add a cast :real(32) to avoid this warning
+int8toReal32.chpl:24: warning: potentially surprising implicit conversion from 'int(16)' to 'real(32)'
+int8toReal32.chpl:24: note: add a cast :real(32) to avoid this warning
+int8toReal32.chpl:25: warning: potentially surprising implicit conversion from 'uint(16)' to 'real(32)'
+int8toReal32.chpl:25: note: add a cast :real(32) to avoid this warning
```

Comment:
We can tune the warning to not fire for assignment. The short way to do that is to handle it directly (i.e. adjust the warning to not fire for assignment calls). But, more broadly, we could implement the warning differently to be more focused on the case where the implicit conversion is leading to a difference in the result type from a set of overloads. I'm just not sure I will have that done by tomorrow.

Reply:
Focusing on assignment would definitely help with this specific case, but trying to generalize it to more of a general rule definitely seems more attractive.

Thinking about whether "no change in result type" is the right discriminator...  If we had:

```chapel
proc foo(x: real(64)) {
  writeln("In 64-bit version");
}

proc foo(x: real(32)) {
  writeln("In 32-bit version");
}
```

(effectively a non-returning, but side-effecting variant on the `sqrt()` case that led to this), would we want that to warn for `foo(myInt8)` or not?  Seems like the answer might be "Maybe, if the assumption is that when adding a `real(16)`, the author would want to add an overload for that as well"?

Maybe more compelling,

```chapel
myReal64 = myOtherReal64 * myInt8;
```

also would be nice not to require a cast for, yet `*` is a returning operator.  But that starts to take me to the other cases I commented on, so let me go see what you wrote there.

Reply:
Slight correction after reading those: you've reminded me that there won't be a warning for conversions to `myReal64`.  But there would be for:

```chapel
myReal32Or64 = myOtherReal32 * myInt8;
```

is that right?  But that feels like a case where we shouldn't warn about the conversion, but do return something.

Reply:
Regarding the other point with your `foo` example, I agree that the warning might be appropriate there; so perhaps the return type is not relevant. However, we could still adjust the warning to be aware of whether or not it mattered for which overload is selected. The warning I implemented is much simpler as an implementation expedience.

Reply:
> we could still adjust the warning to be aware of whether or not the implicit conversion mattered for which overload is selected.

Maybe that's what I've been grasping for...

Reply:
Yes, I expect that for `myOtherReal32 * myInt8`, we could have a different implementation of the warning that is more aware of overload resolution, and that would not fire in that case (because we're going to pick the `real(32)` `*` due to `myOtherReal32` & what `real(t)` `myInt8` would prefer to implicitly convert to is irrelevant).